research/train.py

```diff
@@ -0,0 +1,42 @@
+import argparse
+import torch
+import json
+from time import time
+from steganogan import SteganoGAN
+from steganogan.loader import DataLoader
+from steganogan.critics import BasicCritic
+from steganogan.decoders import DenseDecoder
+from steganogan.encoders import BasicEncoder, ResidualEncoder, DenseEncoder
+
+torch.manual_seed(42)
+timestamp = int(time())
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--epochs', default=32, type=int)
+parser.add_argument('--encoder', default="basic", type=str)
+parser.add_argument('--data_depth', default=1, type=int)
+parser.add_argument('--hidden_size', default=32, type=int)
+parser.add_argument('--dataset', default="div2k", type=str)
+args = parser.parse_args()
+
+train = DataLoader('data/%s/train/' % args.dataset)
```

Comment:
Please use `os.path.join` instead of building the path as a string.
Also, for string formatting, please use the new style `'{}'.format(value)`.